### PR TITLE
feat:Encrypt user passwords on db

### LINKED ISSUE #53: Encrypt user passwords on db
We need to write and read password on db with encryption way. See [this microservice](https://github.com/bojidaryovchev/nest-angular/blob/master/src/server/utilities/encryption.ts) as reference.

### PR DESCRIPTION
**Pull request by Luancarlos**
Implementing password encryption.
Encryption made with the bcrypt library
https://www.npmjs.com/package/bcrypt
Close #53.

### CODE REVIEW

**Comment by amanganiello90** [comment]
Thanks for the contribution!  But we have to fix unit test and test integration:
Check the log: https://github.com/jhipster/generator-jhipster-nodejs/pull/151/checks?check_run_id=843838852

The error is:
```
Error: Copying template server/src/utilities/encryption.ts.ejs failed. [Error: ENOENT: no such file or directory, open '/home/runner/work/generator-jhipster-nodejs/generator-jhipster-nodejs/generators/server/templates/server/src/utilities/encryption.ts.ejs']
```

File: generators/server/templates/server/package.json.ejs [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "@nestjs/platform-express": "6.5.3",
     "@nestjs/swagger": "3.1.0",
     "@nestjs/typeorm": "6.1.3",
+    "bcrypt": "^5.0.0",
```
Comment (amanganiello90) [inline generators/server/templates/server/package.json.ejs:37]: Remove also ^ token

File: generators/server/templates/server/src/service/user.service.ts.ejs [hunk 2]
```diff
@@ -37,6 +38,7 @@ export class UserService {
 
   async save(user: User): Promise<User | undefined> {
     user = this.convertInAuthorities(user);
+    user.password = await generateHashedPassword(user.password);
```
Comment (amanganiello90) [inline generators/server/templates/server/src/service/user.service.ts.ejs:41]: Maybe you have to move this encrypt method to the user.repository class in save methods.
Then, you have to add a decrypt method in findOne repository method. The reason is that the integration tests for security and entity fail as result of not decrypt in the rest get api

Comment (Luancarlos) [reply generators/server/templates/server/src/service/user.service.ts.ejs:41]: ok, I will do as corrections

Comment (Luancarlos) [reply generators/server/templates/server/src/service/user.service.ts.ejs:41]: I will make corrections

**Comment by amanganiello90** [comment]
I replace this PR with https://github.com/jhipster/generator-jhipster-nodejs/pull/152

**Pull request closed without merging**